modules/jupyter/authentication/authenticator/gcpiapjwtauthenticator/gcpiapjwtauthenticator.py

```diff
@@ -59,33 +95,48 @@ class GCPIAPAuthenticator(Authenticator):
     header_name = Unicode(
         config=True,
         help="""HYYP header to inspect for the authenticated JWT.""")
-    
+
     cookie_name = Unicode(
         config=True,
         help="""The name of the cookie field used to specify the JWT token""")
 
     param_name = Unicode(
         config=True,
         help="""The name of the query parameter used to specify the JWT token""")
-    
-    expected_audience = Unicode(
+
+    project_id = Unicode(
```

Comment:
Is there a reason we are adding project_id, project_number, and namespace? Will we need them and will the JWT tokens contain these fields?

Reply:
expected_audiences = [f"/projects/{self.authenticator.project_number}/global/backendServices/{service_id}" for service_id in gcp_backend_services_ids]